### PR TITLE
egltrace: implement eglSetDamageRegion/eglSwapBuffersWithDamage on retrace

### PR DESCRIPTION
**Pull request by enunes**
These seem to be functional, although I'm not sure if this is the correct way to implement it in apitrace. Just let me know if something can be improved.

### CODE REVIEW

File: retrace/glws.hpp [hunk 1]
```diff
@@ -148,11 +148,15 @@ class Drawable
...
 
+    virtual void setDamageRegion(int *rects, int nrects) {};
+
     virtual void setName(const char *name) {}
 
     virtual void copySubBuffer(int x, int y, int width, int height);
 
     virtual void swapBuffers(void) = 0;
+
+    virtual void swapBuffersWithDamage(int *rects, int nrects) {};
```
Comment (jrfonseca) [inline retrace/glws.hpp:159]: This function should default to call `swapBuffers`.

`;` should be removed.

File: retrace/glws.hpp [hunk 2]
```diff
@@ -148,11 +148,15 @@ class Drawable
         visible = true;
     }
 
+    virtual void setDamageRegion(int *rects, int nrects) {};
```
Comment (jrfonseca) [inline retrace/glws.hpp:151]: `;` should be removed.

File: retrace/glws_egl_xlib.cpp [hunk 3]
```diff
@@ -129,6 +129,13 @@ class EglDrawable : public Drawable
         eglWaitNative(EGL_CORE_NATIVE_ENGINE);
     }
 
+    void
+    setDamageRegion(int *rects, int nrects) override {
+        EGLint value;
+        eglQuerySurface(eglDisplay, surface, EGL_BUFFER_AGE_KHR, &value);
```
Comment (jrfonseca) [inline retrace/glws_egl_xlib.cpp:135]: You should check here that the respective EGL extension is supported, and do nothing if not.

File: retrace/glws_egl_xlib.cpp [hunk 4]
```diff
@@ -218,6 +225,12 @@ class EglDrawable : public Drawable
         eglSwapBuffers(eglDisplay, surface);
         processKeys(window);
     }
+
+    void swapBuffersWithDamage(int *rects, int nrects) override {
+        bindAPI(api);
```
Comment (jrfonseca) [inline retrace/glws_egl_xlib.cpp:230]: Likewise, here one should check if the EGL extension is supported, and do nothing otherwise.

**Review by jrfonseca — COMMENTED** [review]
You also need to look at the build failures. For WGL/CGL/etc, the best thing to do is to merely call standard swapBuffers.

Otherwise looks reasonable to me.

**Comment by enunes** [comment]
All of those should be fixed now. I think WGL/CGL/etc are not affected since these are for EGL only. Apparently the CI errors on the windows build were related to the variable sized array.

**Comment by enunes** [comment]
Ping, could you take another look at this one?

**Comment by jrfonseca** [comment]
Looks good.  Thanks

**Pull request merged**